Dockerfile

```diff
@@ -1,50 +1,57 @@
-FROM ubuntu:18.04
+# === BUILDER IMAGE ===
+FROM alpine:3.12 as builder
 USER root
-
 WORKDIR /asn1_codec
-
 VOLUME ["/asn1_codec_share"]
 
-# Add build tools.
-RUN apt-get update && apt-get install -y software-properties-common wget git make gcc-7 g++-7 gcc-7-base && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 100 && update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 100
-
-# Install cmake.
-RUN apt-get install -y cmake
+# update the package manager
+RUN apk update
 
-# install libtool and automake
-RUN apt-get install -y automake libtool
-
-# Install librdkafka.
-RUN apt-get install -y sudo
-RUN wget -qO - https://packages.confluent.io/deb/7.3/archive.key | sudo apt-key add -
-RUN add-apt-repository "deb [arch=amd64] https://packages.confluent.io/deb/7.3 stable main"
-RUN add-apt-repository "deb https://packages.confluent.io/clients/deb $(lsb_release -cs) main"
-RUN apt update
-RUN apt-get install -y libsasl2-modules libsasl2-modules-gssapi-mit libsasl2-dev libssl-dev 
-RUN apt install -y librdkafka-dev
+# add build dependencies
+RUN apk add --upgrade --no-cache --virtual .build-deps \
+    cmake \
+    g++ \
+    make \
+    bash \
+    automake \
+    libtool \
+    autoconf \
+    librdkafka \
+    librdkafka-dev \
+    flex \
+    bison
 
 # Install pugixml
 ADD ./pugixml /asn1_codec/pugixml
 RUN cd /asn1_codec/pugixml && mkdir -p build && cd build && cmake .. && make && make install
 
 # Build and install asn1c submodule
 ADD ./asn1c /asn1_codec/asn1c
-RUN cd /asn1_codec/asn1c && aclocal && test -f configure || autoreconf -iv && ./configure && make && make install
+RUN cd asn1c && test -f configure || autoreconf -iv && ./configure && make && make install
 
-# Generate ASN.1 API.
+# Make generated files available to the build & compile example
 RUN export LD_LIBRARY_PATH=/usr/local/lib
 ADD ./asn1c_combined /asn1_codec/asn1c_combined
-ADD ./scms-asn1 /asn1_codec/scms-asn1
```

Comment:
Why is this removed?

Reply:
We are now including the SCMS files directly in the repository (in a subdirectory of the `asn1c_combined` directory), rather than separately in the scms-asn1 repository. The scms-asn1 git submodule was removed.